src/adapter/src/coord/sequencer/inner/create_view.rs

```diff
@@ -60,17 +58,23 @@ impl Coordinator {
                 return_if_err!(validity.check(self.catalog()), ctx);
             }
 
-            (ctx, stage) = match stage {
+            (ctx, stage, span) = match stage {
                 Validate(stage) => {
+                    let span = span.entered();
                     let next = return_if_err!(self.create_view_validate(ctx.session(), stage), ctx);
-                    (ctx, CreateViewStage::Optimize(next))
+                    (ctx, CreateViewStage::Optimize(next), span.exit())
                 }
                 Optimize(stage) => {
-                    self.create_view_optimize(ctx, stage, otel_ctx).await;
+                    self.create_view_optimize(ctx, stage, span.clone())
```

Comment:
Cloning span and then using it in .instrument in the next line feels like a smell and there's a better way. Suggestions?

Reply:
See comment below, I think only passing it to `.instrument(...)` should be enough